pkg/volume/host_path/host_path.go

```diff
@@ -99,14 +100,22 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
 
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path = filepath.Join("/rootfs", path)
```

Comment:
Yeah. What I meant there was:

1. either break out -- Use `nsenter` to break into docker's network namespace and run appropriate code
2. be rootfs aware -- Use a chroot and assume that mount propagation is correctly configured for the `/rootfs` bindmount.

Number 2 will miss some cases probably, but depending on how the kubelet is run 2 can be enough.

I think 1 is obviously the better option though.